server/controllers/Auth_controller.js

```diff
@@ -2,13 +2,53 @@ import  jwt from 'jsonwebtoken';
 import  dotenv from 'dotenv';
 import  bcrypt from 'bcryptjs';
 import  db from '../models/user_modal';
+import validate from '../helpers/Auth_validation';
+import jwtValidation from '../helpers/jwt'
 import  message from '../helpers/message'
 import {loginValidation,signupValidation} from "../helpers/joi";
 dotenv.config();
 class Auth
 {
 
 
+  signup = async (req,res) =>
+  {
+     
+    const user_email=await db.verify_email(req.body.email);
+  
+    if(user_email){ return message.error(res,409,"email already exist ") }
```

Comment:
@ramadhanbridge , this should not be done in the controller, this should be a middleware